language-plutus-core/plutus-exe/Main.hs

```diff
@@ -58,24 +59,24 @@ evalOpts :: Parser EvalOptions
 evalOpts = EvalOptions <$> input
 
 -- | Parse a program and run it using the CK machine.
-parseRunCk :: BSL.ByteString -> Either (PC.ParseError PC.AlexPosn) CK.CkEvalResult
+parseRunCk :: (MonadError (PC.Error PC.AlexPosn) m) => BSL.ByteString -> m CK.CkEvalResult
 parseRunCk = fmap (CK.runCk . void) . PC.parseScoped
 
 -- | Parse a program and typecheck it.
-parseTypecheck :: BSL.ByteString -> Either (PC.Error PC.AlexPosn) (PC.Type PC.TyNameWithKind ())
-parseTypecheck bs = runExcept $ PC.runQuoteT $ do
-    parsed <- PC.parseProgramQ bs
+parseTypecheck :: (MonadError (PC.Error PC.AlexPosn) m, PC.MonadQuote m) => BSL.ByteString -> m (PC.Type PC.TyNameWithKind ())
```

Comment:
Might be able to write this as `typecheckProgram 10000 <=< annotateProgram <=< parseProgram`, and in any case I think this function should be in `Language.PlutusCore` or some module that's exported - that way library users would have access :)

Reply:
Yeah, I wasn't sure what prefixes of the pipeline to expose. Probably "parse this" and "parse and typecheck this" are the most common ones.

Reply:
It also runs `checkProgram`, and I coudn't figure out how to get that into a `<=<` pipeline.